bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/logger/LogWriter.java

```diff
@@ -0,0 +1,315 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.homeconnect.internal.logger;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.storage.Storage;
+import org.openhab.binding.homeconnect.internal.client.HomeConnectApiClient;
+import org.openhab.binding.homeconnect.internal.client.HomeConnectSseClient;
+import org.openhab.binding.homeconnect.internal.factory.HomeConnectHandlerFactory;
+import org.slf4j.LoggerFactory;
+import org.slf4j.event.Level;
+import org.slf4j.helpers.FormattingTuple;
+import org.slf4j.helpers.MessageFormatter;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ *
+ * Home Connect logger.
+ *
+ * @author Jonas Brüstel - Initial Contribution
+ */
+@NonNullByDefault
+public class LogWriter {
```

Comment:
Do we really need this, it feels very heavy, what about using the MDC:

http://logback.qos.ch/manual/mdc.html

Could that make sense and do we really need all the context, I don't think bindings should contain code like this.

Reply:
Yea, I didn't want to address this in my first pass review, but I don't think that bindings should create their own logging framework. I realize that you are trying to log to have the binding log to its own file but you should leave it up to the users to configure that through the logback configuration files.

IMO all of the custom logging code should be removed and replaced with the standard slf4j logging.